django/db/models/indexes.py

```diff
@@ -44,21 +44,32 @@ def check_name(self):
             self.name = 'D%s' % self.name[1:]
         return errors
 
-    def create_sql(self, model, schema_editor, using=''):
+    def get_sql_create_params(self, model, schema_editor, using=''):
+        """
+        Wrapper method for generating the set of parameters.
+        """
         fields = [model._meta.get_field(field_name) for field_name, order in self.fields_orders]
         tablespace_sql = schema_editor._get_index_tablespace_sql(model, fields)
         quote_name = schema_editor.quote_name
         columns = [
             ('%s %s' % (quote_name(field.column), order)).strip()
             for field, (field_name, order) in zip(fields, self.fields_orders)
         ]
-        return schema_editor.sql_create_index % {
-            'table': quote_name(model._meta.db_table),
-            'name': quote_name(self.name),
-            'columns': ', '.join(columns),
-            'using': using,
-            'extra': tablespace_sql,
-        }
+        return {'table': quote_name(model._meta.db_table),
+                'columns': ', '.join(columns),
+                'name': quote_name(self.name),
+                'extra': tablespace_sql,
+                'with': '',
+                'using': using}
+
+    def create_sql(self, model, schema_editor, using='', parameters=None, sql_create_index=None):
+        sql_create_index = sql_create_index or schema_editor.sql_create_index
+        sql_parameters = parameters or self.get_sql_create_params(model, schema_editor, using)
+        try:
+            return sql_create_index % sql_parameters
+        except KeyError:
```

Comment:
Mainly because of the `with`-parameter. The chain to create the index somehow involves `_create_index_sql` in base/schema.py where `with` is not part of it. So in essence, the additional template string was an easy way to deal with it, but if possible I can look for ways to get rid of it.